CHANGES.rst

```diff
@@ -8,6 +8,7 @@ Version 0.9.0 (UNRELEASED)
 - Adds new ``WorkspaceRetentionRule`` table to store workspace file retention rules.
 - Adds a limit to the number of times a workflow can be restarted.
 - Changes percentage ranges for calculating the quota health status.
+- Changes ``Workflow.get_workspace_disk_usage`` to calculate disk usage and not use the usage values from database.
```

Comment:
Hmm, I'm not sure whether it is OK to _always_ calculate the workspace disk usage without caching result. This could lead to slow REST API replies to commands like `du -w myoldworkflow` even if those old files didn't change much in years...

Besides, how would this influence commands like `list --include-workspace-size`? Would we do everything live as well? (Here the REST API performance is much more important than for single `du` command.)

I'd say `reana-client du` or system-wide `reana-resource-quota-update` cronjob or `reana-client list --include-workspace-size` are all kinds of commands that could trigger disk usage getter, and then if the result is cached in the database, all those commands could return detected values.

E.g. currently I'm getting `-1` in the size column locally on a fresh installation, even if I ask for one disk usage explicitly before:

```console
$ reana-client du -w root6-roofit-snakemake-kubernetes -s
SIZE      NAME
1556480   .
$ reana-client list --include-workspace-size
NAME                                RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS     SIZE
root6-roofit-snakemake-kubernetes   1            2022-08-08T11:45:57   2022-08-08T11:46:10   2022-08-08T11:46:41   finished   -1
root6-roofit-serial-kubernetes      1            2022-08-08T11:45:24   2022-08-08T11:45:36   2022-08-08T11:45:51   finished   -1
root6-roofit-cwl-kubernetes         1            2022-08-08T11:44:49   2022-08-08T11:45:03   2022-08-08T11:45:18   finished   -1
```

I guess we may want to polish the du/list/cronjob chain and reuse of results more widely?

Reply:
> Besides, how would this influence commands like list --include-workspace-size? Would we do everything live as well? (Here the REST API performance is much more important than for single du command.)

The performance will not be affected. `du` command in this PR doesn't share a code. The `list --include-workspace-size` command uses quotas, and if they do not exist, it returns `-1`. If quotas are disabled and we want to cache results, `du` and `list --include-workspace-size` commands will be outdated. 

> I guess we may want to polish the du/list/cronjob chain and reuse of results more widely?

Yes, we should. Our quota codebase is messy and hard to modify and understand. Should I look into ways to improve it? I can prepare the documentation for the REANA dev wiki describing how current quotas work. From there we can decide what to do.

Reply:
OK, I confirm that `list` is OK. Hence we can merge this fix. And we can create a separate issue for the `-1` bit.  (That we should at the very least document in `reana-client list --help`, and at the very most update so that people can consult disk space even if quotas are disabled.)